build-tools/cmake/xa_macros.cmake

```diff
@@ -77,3 +77,51 @@ macro(xa_common_prepare)
       )
   endif()
 endmacro()
+
+macro(xa_macos_prepare_arm64)
+  if(APPLE)
+    set(SDK_SUPPORTS_ARM64 False)
+    set(SDK_SUPPORTS_X86_64 False)
+    execute_process(
+      COMMAND xcode-select -p
+      RESULT_VARIABLE XCODE_SELECT_RESULT
+      OUTPUT_VARIABLE XCODE_DEVELOPER_PATH
+      )
+    if(NOT ${XCODE_SELECT_RESULT} EQUAL "0")
+      message(WARNING "xcode-select failed with result ${XCODE_SELECT_RESULT}")
```

Comment:
As per the [cmake `execute_process()` docs](https://cmake.org/cmake/help/latest/command/execute_process.html), `RESULT_VARIABLE` is an *integer*, the *exit code* of the process, while `OUTPUT_VARIABLE` is the unified stdout+stderr of the process run.

Question: when `execute_process()` is run, does it *also* print stdout/stderr to *`cmake`s stdout* -- i.e. our AzDO log file -- so that we can see the result of the executed command?

If `xcode-select -p` fails, we'll want to know *why* it failed, which means we'll need access to stdout/stderr.  If `execute_process()` always "forwards" this output, we're fine.  If it *doesn't* "forward" this output, then we need to ensure that this `message()` includes it:

```cmake
message(WARNING "xcode-select failed with result ${XCODE_SELECT_RESULT} output ```${XCODE_DEVELOPER_PATH}```")
```

Reply:
```$ ninja -j1 -v
ninja: warning: bad deps log signature or version; starting over
[0/1] /Users/grendel/android-toolchain/sdk/cmake/3.10.2.4988404/bin/cmake -H/Users/grendel/vc/xamarin/xamarin-android/src/monodroid -B/Users/grendel/vc/xamarin/xamarin-android/src/monodroid/obj/Debug/host-Darwin-Debug
-- BIN_DIR exists
xcode-select: error: invalid argument '-pp'
Usage: xcode-select [options]

Print or change the path to the active developer directory. This directory
controls which tools are used for the Xcode command line tools (for example, 
xcodebuild) as well as the BSD development commands (such as cc and make).

Options:
  -h, --help                  print this help message and exit
  -p, --print-path            print the path of the active developer directory
  -s <path>, --switch <path>  set the path for the active developer directory
  --install                   open a dialog for installation of the command line developer tools
  -v, --version               print the xcode-select version
  -r, --reset                 reset to the default command line tools path
CMake Warning at /Users/grendel/vc/xamarin/xamarin-android/build-tools/cmake/xa_macros.cmake:91 (message):
  xcode-select failed with result 1
Call Stack (most recent call first):
  CMakeLists.txt:95 (xa_macos_prepare_arm64)
```